### PR TITLE
Vendor in case functions

### LINKED ISSUE #2464: Vendor in `case` utilities
## Problem

We are using `case` package for `title`, `camel` and `kebab` functions. This module is CJS-only, and to Migrate to Nuxt 3 we need to move away from CJS modules.

## Description

Extract the `title`, `camel` and `kebab` functions from [`case` package](https://github.com/nbubna/Case/blob/b61c98d173cf8080b05181e323ef27af8249388b/src/Case.js) into `/frontend/src/utils/case.ts`. Replace the imports throughout the app: https://github.com/search?q=repo%3AWordPress%2Fopenverse+%22from+%5C%22case%5C%22%22&type=code

## Alternatives

One alternative would be to finish the ESM-conversion PR in the `case` package. However, it seems to require some familiarity with packaging quirks.
Another alternative would be to use Vite workaround: (instead of `const { kebab } = require "case"` use `import casePkg; const { kebab } = casePkg;`.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2464 by @obulat

## Description

This PR vendors in the implementation for `case` functions that we use (camelCase, capitalCase, titleCase) from `change-case` JS package.

In the locales scripts files, I could not use the `camelCase` function because those scripts are called by `node`, not transpiled, so they cannot import anything from `.ts` files. I added a simpler version in JS, `kebabToCamel`, because we convert only kebab cased values, and do not need the complex splitting used in `camelCase` function.

## Testing Instructions

The CI should pass. 

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
@obulat CI is passing on the PR and the changes look good too, why is this a draft?

**Review by dhruvkb — APPROVED** [review]
Unfortunate circumstances but execution LGTM!

File: frontend/src/utils/case.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// Copied from https://github.com/blakeembrey/change-case
```
Comment (AetherUnbound) [inline frontend/src/utils/case.ts:1]: Do you mind adding an explanation here as to why it was copied, similar to what's in the issue description?

**Review by AetherUnbound — COMMENTED** [review]
This looks fine, but should we also vendor the tests associated with these utilities to ensure functionality?

**Comment by obulat** [comment]
> This looks fine, but should we also vendor the tests associated with these utilities to ensure functionality?

👍  I added the tests.

**Review by AetherUnbound — APPROVED** [review]
LGTM!

**Pull request merged**